packages/flutter/lib/src/material/material_state.dart

```diff
@@ -45,62 +47,16 @@ import 'input_border.dart';
 ///    `MaterialStateProperty` which is used in APIs that need to accept either
 ///    a [TextStyle] or a [MaterialStateProperty<TextStyle>].
 /// {@endtemplate}
-enum MaterialState {
-  /// The state when the user drags their mouse cursor over the given widget.
-  ///
-  /// See: https://material.io/design/interaction/states.html#hover.
-  hovered,
-
-  /// The state when the user navigates with the keyboard to a given widget.
-  ///
-  /// This can also sometimes be triggered when a widget is tapped. For example,
-  /// when a [TextField] is tapped, it becomes [focused].
-  ///
-  /// See: https://material.io/design/interaction/states.html#focus.
-  focused,
-
-  /// The state when the user is actively pressing down on the given widget.
-  ///
-  /// See: https://material.io/design/interaction/states.html#pressed.
-  pressed,
-
-  /// The state when this widget is being dragged from one place to another by
-  /// the user.
-  ///
-  /// https://material.io/design/interaction/states.html#dragged.
-  dragged,
-
-  /// The state when this item has been selected.
-  ///
-  /// This applies to things that can be toggled (such as chips and checkboxes)
-  /// and things that are selected from a set of options (such as tabs and radio buttons).
-  ///
-  /// See: https://material.io/design/interaction/states.html#selected.
-  selected,
-
-  /// The state when this widget overlaps the content of a scrollable below.
-  ///
-  /// Used by [AppBar] to indicate that the primary scrollable's
-  /// content has scrolled up and behind the app bar.
-  scrolledUnder,
-
-  /// The state when this widget is disabled and cannot be interacted with.
-  ///
-  /// Disabled widgets should not respond to hover, focus, press, or drag
-  /// interactions.
-  ///
-  /// See: https://material.io/design/interaction/states.html#disabled.
-  disabled,
-
-  /// The state when the widget has entered some form of invalid state.
-  ///
-  /// See https://material.io/design/interaction/states.html#usage.
-  error,
-}
+typedef MaterialState = WidgetState;
 
 /// Signature for the function that returns a value of type `T` based on a given
 /// set of states.
-typedef MaterialPropertyResolver<T> = T Function(Set<MaterialState> states);
+///
+/// See also:
+///
+///  * [WidgetPropertyResolver], the non-Material form of `MaterialPropertyResolver`
+///    that is inherited by `MaterialPropertyResolver.
```

Comment:
inherited? I think this should probably say what you said on the other one that they can just be used interchangable?